packages/astro/src/runtime/server/render/component.ts

```diff
@@ -473,6 +477,18 @@ export async function renderComponent(
 	return await renderFrameworkComponent(result, displayName, Component, props, slots);
 }
 
+function normalizeProps(props: Record<string, any>): Record<string, any> {
+	if ('class:list' in props) {
```

Comment:
```suggestion
	if (props['class:list'] !== undefined) {
```

Checked the benchmarks, `props['class:list'] !== undefined` is _slightly_ faster than `typeof props['class:list'] !== 'undefined'`